packages/labs/task/src/task.ts

```diff
@@ -30,11 +30,11 @@ export const initialState = Symbol();
 
 export type TaskStatus = typeof TaskStatus[keyof typeof TaskStatus];
 
-export type StatusRenderer<R> = {
-  initial?: () => unknown;
-  pending?: () => unknown;
-  complete?: (value: R) => unknown;
-  error?: (error: unknown) => unknown;
+export type StatusRenderer<R, Q> = {
```

Comment:
nit: `R` here stands for "result" I think. Rather than use the next letter in the alphabet, let's do `V` for the render value.

Reply:
It'd be very reasonable for the render callbacks to return different types. I'd default the type parameter to `unknown`: `<R, V = unknown>`

Reply:
> It'd be very reasonable for the render callbacks to return different types. I'd default the type parameter to `unknown`: `<R, V = unknown>`

To me it seems like I still get typecheck errors now even with a default also added in the method directly. Maybe that's the reason it's typed as `unknown`. As an alternative I was thinking about `V1,V2,V3,V4` and then the resulting one is the union of them.